### PR TITLE
Improve storage number unpacking by using a lookup table.

### PR DESCRIPTION
**Pull request by vkalintiris**
The LUT contains precomputed values of multipliers/divisors that are
used to unpack storage numbers into calculated numbers efficiently.

##### Summary

Performance improves by at least 35% when testing with random integers
in `uint32_t`s entire range. Unpacking smaller random values (eg. in the
range [0,1000]) is faster by 200-300%.

##### Test Plan

1. Compare calculated_numbers in old vs new version,
2. Run agent and check dashboards for any irregularities.

### CODE REVIEW

**Comment by vkalintiris** [comment]
Benchmark source code: https://gist.github.com/vkalintiris/2fc64b8b444dcbba2216e363d6121a87

File: libnetdata/storage_number/storage_number.c [hunk 1]
```diff
@@ -81,23 +81,40 @@ storage_number pack_storage_number(calculated_number value, uint32_t flags) {
     return r;
 }
 
+// Lookup table to make storage number unpacking efficient.
+static calculated_number lut10x[4 * 8];
+
+__attribute__((constructor)) void initialize_lut(void) {
```
Comment (vkalintiris) [inline libnetdata/storage_number/storage_number.c:87]: I don't think there exists a compiler out there that manages to build netdata successfully without already supporting the constructor attribute. If you don't like this I can make this function visible in a header and call it from main.

**Review by vlvkobal — DISMISSED** [review]

File: libnetdata/storage_number/storage_number.c (hunk 1 above)
Comment (underhood) [reply libnetdata/storage_number/storage_number.c:87]: does this include MS compiler? we plan in future to move to Win

Comment (vkalintiris) [reply libnetdata/storage_number/storage_number.c:87]: `msvc` won't compile this code whereas `clang-cl` will.

Comment (underhood) [reply libnetdata/storage_number/storage_number.c:87]: ok keep it as it is for now I guess. we will deal with this later

**Review by underhood — DISMISSED** [review]

**Review by vlvkobal — APPROVED** [review]

**Review by stelfrag — APPROVED** [review]

**Pull request merged**